upgrade.php

```diff
@@ -23,25 +24,92 @@
 echo "WELCOME TO THE SNIPE-IT UPGRADER! \n";
 echo "--------------------------------------------------------\n\n";
 echo "This script will attempt to: \n\n";
+echo "- validate some very basic .env file settings \n";
 echo "- check your PHP version and extension requirements \n";
+echo "- check directory permissions \n";
 echo "- do a git pull to bring you to the latest version \n";
 echo "- run composer install to get your vendors up to date \n";
 echo "- run migrations to get your schema up to date \n";
 echo "- clear out old cache settings\n\n";
 
+
+
+echo "--------------------------------------------------------\n";
+echo "STEP 1: Checking .env file: \n";
+echo "- Your .env is located at ".getcwd()."/.env \n";
+echo "--------------------------------------------------------\n\n";
+
+
+// Check the .env looks ok
+$env = file('.env');
+$env_error_count = 0;
+
+// Loop through each line of the .env
+foreach ($env as $line_num => $line) {
+
+    if ((strlen($line) > 1) && (strpos($line, "#") !== 0)) {
+
+        list ($env_key, $env_value) = $env_line = explode('=', $line);
```

Comment:
Should this be `explode('=', $line, 2)`? (you want any `=` in the value-side to be part of the `$env_value` I assume)

Reply:
No, the `=` should not be included, it's just grabbing the key and the value as two different variables.

Reply:
I mean a line like this - 

`APP_URL=https://somethingtotallyvalid=someothercrapwhatever`

That split will return three things, the APP_URL, the https://somethingtotallyvalid, and someothercrapwhatever - the last item will be thrown away. I don't know if that's what you want.

I'm also a little weirded out by the `list (blah, blah ) = blah = explode()` line, that seems a little odd to me, but I trust you - if it works, it works.

Reply:
To clarify, in the standard BLAH=value setup, at least for the variables we use here, I don't need or want the `=` sign. We might run into a situation where we have an `=` symbol in an API key, app key, db password, etc - but we're not really introspecting into those values right now.